### ISSUE TITLE
Unable to build 1.1pre17 in mingw64 (msys2)

### ISSUE THREAD

**Issue by Teteros** [gsliepen/tinc#211]
```
  CC       protocol_key.o
protocol_auth.c: In function 'send_chal_reply':
protocol_auth.c:728:48: error: 'struct connection_t' has no member named 'rsa'
  const size_t len = rsa_size(myself->connection->rsa);
                                                ^~
protocol_auth.c:729:21: warning: implicit declaration of function 'digest_length'; did you mean 'filelength'? [-Wimplicit-function-declaration]
  size_t digestlen = digest_length(c->indigest);
                     ^~~~~~~~~~~~~
                     filelength
protocol_auth.c:729:36: error: 'connection_t' {aka 'struct connection_t'} has no member named 'indigest'
  size_t digestlen = digest_length(c->indigest);
                                    ^~
protocol_auth.c:734:6: warning: implicit declaration of function 'digest_create'; did you mean 'list_free'? [-Wimplicit-function-declaration]
  if(!digest_create(c->indigest, c->mychallenge, len, digest)) {
      ^~~~~~~~~~~~~
      list_free
protocol_auth.c:734:21: error: 'connection_t' {aka 'struct connection_t'} has no member named 'indigest'
  if(!digest_create(c->indigest, c->mychallenge, len, digest)) {
                     ^~
  CC       protocol_misc.o
make[2]: *** [Makefile:1053: protocol_auth.o] Error 1
make[2]: *** Waiting for unfinished jobs....
  CC       protocol_subnet.o
protocol_key.c:37:13: warning: 'mykeyused' defined but not used [-Wunused-variable]
 static bool mykeyused = false;
             ^~~~~~~~~
make[2]: Leaving directory '/c/Program Files/tinc/mingw-w64-tinc-pre/src/tinc/src'
make[1]: *** [Makefile:389: all-recursive] Error 1
```

(previous release 1.1pre16 does build)
`gcc --version`
gcc.exe (Rev3, Built by MSYS2 project) 8.2.0

**Comment by Teteros** [comment]
Looks like this isn't msys2 environment specific, it happens on linux when `--disable-legacy-protocol` is passed to configure as well.

**Comment by gsliepen** [comment]
Thanks for reporting this issue! It's fixed in 1ebadf1.
